keycloak/bin/add-users-by-email.sh

```diff
@@ -0,0 +1,71 @@
+#! /usr/bin/env bash
+#
+# Add new users to osmp realm based on file of newline-separated emails.
+# Each line must end with a newline.
+# uses client_credentials flow by default. add --password flag to use resource owner password flow
+
+# Example usage with Docker:
+# cat emails.txt | docker exec -i <keycloak_container> bash /usr/local/bin/add-users-by-email.sh [--password]
+
+set -euo pipefail
+
+granttype="${AUTH0_GRANT_TYPE}"
+
+export PATH=$PATH:/opt/jboss/keycloak/bin
+kcadm.sh config credentials --server http://localhost:8080/auth --realm master --user "${KEYCLOAK_USER}" --password ${KEYCLOAK_PASSWORD}
+
+# Get management API access token (assuming it's Auth0)
+if [ "$granttype" = "password" ]; then
+    tokenresponse=$(curl --request POST \
+        --url "${AUTH0_BROKER_TOKEN_URL}" \
+        --header "Content-Type: application/x-www-form-urlencoded" \
+        --data "client_id=${AUTH0_BROKER_CLIENT_ID}" \
+        --data "grant_type=password" \
+        --data "username=${AUTH0_BROKER_USERNAME}" \
+        --data "password=${AUTH0_BROKER_PASSWORD}" \
+        --data "audience=${AUTH0_BROKER_MANAGEMENT_API_URL}"
+    )
+else
+    tokenresponse=$(curl --request POST \
+        --url "${AUTH0_BROKER_TOKEN_URL}" \
+        --header "Content-Type: application/x-www-form-urlencoded" \
+        --data "client_id=${AUTH0_BROKER_CLIENT_ID}" \
+        --data "grant_type=client_credentials" \
+        --data "client_secret=${AUTH0_BROKER_CLIENT_SECRET}" \
+        --data "audience=${AUTH0_BROKER_MANAGEMENT_API_URL}"
+    )
+fi
+
+accesstoken=$(
+    echo "${tokenresponse}" | grep -zoP '"access_token":\s*"\K[^\s,]*(?="\s*,)'
```

Comment:
Are you trying to parse JSON with grep?

Reply:
Yep. The regex matches the value inside the double quotes whose key is "access_token". Using something like [jq](https://stedolan.github.io/jq/) would probably be better, but it's not included in the keycloak container by default.

Reply:
This script bin is only mounted in development. Is this a script to be used in development, or run from a development machine? If so, we can write this script to assume `jq` exists (heck, even Node.js) and get more reliable parsing there, and use `docker-compose exec keycloak` for the calls to kcadm.

Reply:
My understanding was that this script would be used to semi-automate the process of adding G4RD users to OSMP Keycloak in production. I suppose it does make more sense then to fix the script to run outside of Docker by some trusted admin user in production. I'll rework the script.